File: client/src/app/services/chat-service/chat-display.service.ts
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';

@Injectable({
    providedIn: 'root',
})
export class ChatDisplayService {
    isRoomSelected: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
    isSearchSelected: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
    isChatVisible: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
    selectRoom() {
        this.isRoomSelected.next(true);
    }
    selectSearch() {
        this.isSearchSelected.next(true);
    }
    deselectRoom() {
        this.isRoomSelected.next(false);
    }
    deselectSearch() {
        this.isSearchSelected.next(false);
    }
    toggleChat() {
        this.isChatVisible.next(!this.isChatVisible.value);
    }

    hideChat() {
        if (this.isChatVisible.value) {
            this.isChatVisible.next(false);
        }
    }

    reset() {
        this.isRoomSelected.next(false);
        this.isSearchSelected.next(false);
        this.isChatVisible.next(false);
    }
}

File: flutter/app/lib/domain/services/chat_display_service.dart
import 'dart:async';

import 'package:rxdart/rxdart.dart';

class ChatDisplayService {
  final isRoomSelected = BehaviorSubject<bool>.seeded(false);
  final isSearchSelected = BehaviorSubject<bool>.seeded(false);
  final isChatVisible = BehaviorSubject<bool>.seeded(false);

  Stream<bool> get isRoomSelectedStream => isRoomSelected.stream;
  Stream<bool> get isSearchSelectedStream => isSearchSelected.stream;
  Stream<bool> get isChatVisibleStream => isChatVisible.stream;

  // void selectRoom() {
  //   isRoomSelected.add(true);
  // }

  void selectSearch() {
    isSearchSelected.add(true);
  }

  // void deselectRoom() {
  //   isRoomSelected.add(false);
  // }

  void deselectSearch() {
    isSearchSelected.add(false);
  }

  void toggleChat() {
    isChatVisible.add(!isChatVisible.value);
  }

  void deselectRoom() {}

  // void dispose() {
  //   isRoomSelected.close();
  //   isSearchSelected.close();
  //   isChatVisible.close();
  // }
}
